client/rest/src/main/java/org/elasticsearch/client/RuntimeInfo.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+class RuntimeInfo {
+
+    /**
+     * Returns runtime info to append to the metadata header. Currently looks up classes identifying non-Java JVM
+     * languages and appends their major/minor version patch.
+     */
+    public static String getRuntimeMetadata() {
+        StringBuilder s = new StringBuilder();
+        String version;
+
+        version = HlrcKind();
+        if (version != null) {
+            s.append(",hl=").append(version);
+        }
+
+        version= kotlinVersion();
```

Comment:
Do any of these version functions need to handle pre-release information?

Reply:
Service/language/http-client versions are not stripped, and so can include pre-release information. For language runtimes we only keep major.minor as this information allows us to better understand the ecosystem and the patch version doesn't bring useful information. The patch version for these may only be of interest if we build dedicated clients for these languages, in which case their version will be reported as the language version.